vllm/model_executor/layers/linear.py

```diff
@@ -286,6 +287,45 @@ def __init__(
         self.tp_rank = get_tensor_model_parallel_rank() if not disable_tp else 0
         self.tp_size = get_tensor_model_parallel_world_size() if not disable_tp else 1
 
+        if (
+            self.quant_method.__class__.__name__ in WEIGHT_LOADER_V2_SUPPORTED
+            and callable(getattr(self, "weight_loader_v2", None))
+        ):
+            self.weight_loader_v2 = self.add_preprocessor_to_weight_loader(
+                self.weight_loader_v2  # type: ignore
+            )
+        if callable(getattr(self, "weight_loader", None)):
+            self.weight_loader = self.add_preprocessor_to_weight_loader(
+                self.weight_loader  # type: ignore
+            )
+
+    def add_preprocessor_to_weight_loader(self, weight_loader: Callable) -> Callable:
+        def weigth_loader_with_preprocessor(
+            param: Parameter | BasevLLMParameter,
+            loaded_weight: torch.Tensor,
+            *args,
+            **kwargs,
+        ):
+            if self.quant_method is None:
+                return weight_loader(param, loaded_weight, *args, **kwargs)
+            results = self.quant_method.process_weights_before_loading(
```

Comment:
This doesn't seem like it would work if a layer wants to load parameters that aren't weights, for example, bias terms.

Reply:
Weight loaders are used to load all parameters belonging to a module, there isn't an easy way for a quant_config to choose which `process_weights_before_loading` function gets called for which parameter via the weight loaders.

You could do something like hold a reference to the parameter to check against, but that's pretty complicated, and it'd have to be a weak ref or else you could run into gc issues.